src/main/java/seedu/dietbook/calculator/CalculatorData.java

```diff
@@ -0,0 +1,133 @@
+package seedu.dietbook.calculator;
+
+import seedu.dietbook.food.Food;
+import seedu.dietbook.list.FoodList;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculatorData {
+    FoodList list;
+    List<Food> foods = null;
+
+    public CalculatorData() {
+        this.list = new FoodList();
+    }
+
+    public CalculatorData(FoodList foodList) {
+        this.list = foodList;
+    }
+
+    public void inputData(FoodList foodList) {
+        this.list = foodList;
+    }
+
+    public List<Integer> getTotalCalorie() {
+        this.foods = list.getFoods();
```

Comment:
I suspect that the method called should be `getPortionedFoods()`

`getFoods()` does not take into consideration the portion size provided by the user and it seems that you did not get and handle the portion sizes separately. Using `getPortionedFoods()` will return a food item with scaled nutrional values (different from the item the user entered)